api/base/exceptions.py

```diff
@@ -3,16 +3,42 @@
 from rest_framework import status
 from rest_framework.exceptions import APIException, ParseError
 
+def dict_error_formatting(error):
+    """
+    Formats all dictionary error messages for both single and bulk requests
+    """
+
+    formatted_error_list = []
+    # Error objects may have the following members. Title and id removed to avoid clash with "title" and "id" field errors.
+    top_level_error_keys = ['links', 'status', 'code', 'detail', 'source', 'meta']
+
+    # Resource objects must contain at least 'id' and 'type'
+    resource_object_identifiers = ['type', 'id']
+
+    for error_key, error_description in error.iteritems():
+        if isinstance(error_description, basestring):
+            error_description = [error_description]
+
+        if error_key in top_level_error_keys:
+            formatted_error_list.append({error_key: error_description[0]})
```

Comment:
This seems to only add the first error message for `error_key`; what if there are multiple errors?